AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/ui/user/UserViewModel.kt

```
@@ -87,8 +88,28 @@ open class UserViewModel(
       }
     }
   }
+
   /** Sets the current user. */
   fun setUser(user: User) {
     _user.value = user
   }
+
+  /** Updating the officeId when creating or joining an office */
+  fun updateVetOfficeId(officeId: String?) {
+    viewModelScope.launch {
+      val current = _user.value
+
+      // Only vets should get an officeId
+      if (current is Vet) {
+        val updated = current.copy(officeId = officeId)
+
+        try {
+          userRepository.updateUser(updated)
+          _user.value = updated
+        } catch (e: Exception) {
+          Log.e("UserViewModel", "Failed to update vet officeId", e)
```

catching this error without propagating it is dangerous, you should throw the error back when this fails.